lib/fluent/plugin/prometheus.rb

```diff
@@ -56,6 +56,68 @@ def self.parse_labels_elements(conf)
         base_labels
       end
 
+      def self.parse_initlabels_elements(conf, base_labels)
```

Comment:
How about removing the prefix `self.` to make it as instance method?
So that you can use `fluentd_worker_id` directly instead of setting `@worker_id` via `Fluent::Plugin::Prometheus.set_worker_id`.

Reply:
Hello,
Seems to me the new method `parse_initlabels_elements` has to be called like `parse_labels_elements` in `Metric` class in its `initialize` method (around line 230), hence it is a class method we need not an instance method
And yes from the moment it is a class method we need this (dirty) trick to pass down the `fluentd_worker_id` :/

Do you see a way of writing that without droping support for `fluentd_worker_id` or having to reorganize the code?

Reply:
Ah, I see. Sorry I overlooked it.

> Do you see a way of writing that without droping support for fluentd_worker_id or having to reorganize the code?

I think passing it by argument is proper way, but such change seems overkill in this case.

The worker ID is determined per process, can be retrieved from `ENV['SERVERENGINE_WORKER_ID']`.

https://github.com/fluent/fluentd/blob/c0f1db966a2809a13de217b0ec0adc9a29decc79/lib/fluent/plugin/base.rb#L51

```ruby
      def fluentd_worker_id
        return @_fluentd_worker_id if @_fluentd_worker_id
        @_fluentd_worker_id = (ENV['SERVERENGINE_WORKER_ID'] || 0).to_i
        @_fluentd_worker_id
      end
```

I think using `ENV['SERVERENGINE_WORKER_ID']` is enough for this situation.

Or, using [`Fluent::VariableStore`](https://github.com/fluent/fluentd/blob/c0f1db966a2809a13de217b0ec0adc9a29decc79/lib/fluent/variable_store.rb
) instead of class variable or class-instance variable is recommended in fluentd plugins.

Reply:
I liked the `ENV['SERVERENGINE_WORKER_ID']` idea, I replaced by this
Let me know if you see anything else to change